conda_build/convert.py

```diff
@@ -8,517 +8,701 @@
+    executable_directory (str) -- the file path to temporary directory's 'Scripts' directory
+    executable (str) -- the filename of the script to add to paths.json
+    """
+    with open('{}/{}' .format(executable_directory, executable), 'rb') as script_file:
+        script_file_contents = script_file.read()
+        new_path = {"_path": "Scripts/{}" .format(executable),
+                    "path_type": "hardlink",
+                    "sha256": hashlib.sha256(script_file_contents).hexdigest(),
+                    "size_in_bytes": os.path.getsize('{}/{}' .format(
+                        executable_directory, executable))
+                    }
+    return new_path
+
+
+def update_paths_file(temp_dir, target_platform):
+    """Update the paths.json file when converting between platforms.
+
+    Positional arguments:
+    temp_dir (str) -- the file path to the temporary directory containing the source
+        package's extracted contents
+    target_platform (str) -- the platform to target: 'unix' or 'win'
     """
+    paths_file = '{}/info/paths.json' .format(temp_dir)
+
+    if os.path.isfile(paths_file):
+        with open(paths_file) as file:
+            paths = json.load(file)
+
+        if target_platform == 'win':
+            for path in paths['paths']:
+                if path['_path'].startswith('lib'):
+                    path['_path'] = update_lib_path(path['_path'], 'win')
+
+                elif path['_path'].startswith('bin'):
+                    path['_path'] = update_executable_path(path['_path'], 'win')
+
+            script_directory = '{}/Scripts/' .format(temp_dir)
+            if os.path.isdir(script_directory):
+                for script in os.listdir(script_directory):
+                    if script.endswith('.exe'):
+                        paths['paths'].append(add_new_windows_path(script_directory, script))
+
+        elif target_platform == 'unix':
+            for path in paths['paths']:
+                if path['_path'].startswith('Lib'):
+                    path['_path'] = update_lib_path(path['_path'], 'unix', temp_dir)
+
+                elif path['_path'].startswith('Scripts'):
+                    path['_path'] = update_executable_path(path['_path'], 'unix')
+
+                if path['_path'].endswith(('.bat', '.exe')):
+                    paths['paths'].remove(path)
+
+        with open(paths_file, 'w') as file:
+            json.dump(paths, file)
+
+
+def retrieve_executable_name(executable):
+    """Retrieve the name of the executable to rename.
+
+    When converting between unix and windows, we need to be careful
+    that the executables are renamed without their file extensions.
+
+    Positional arguments:
+    executable (str) -- the executable to rename including its file extension
+    """
+    return os.path.splitext(os.path.basename(executable))[0]
+
+
+def rename_executable(directory, executable, target_platform):
+    """Rename an executable file when converting between platforms.
```

Comment:
I'm a little wary on this one.  Files in the bin folder can binaries, right?  Do we have a way of distinguishing them from things that are being used as entry point scripts, or just scripts (both in the setuptools sense of the word).  entry_points are things that point at a particular python function, where scripts are files that get run (also by the same exe's on Windows that entry points use).